src/main/java/com/facebook/openwifirrm/ucentral/models/AggregatedState.java

```diff
@@ -0,0 +1,221 @@
+			if (rate == null || rate.chWidth != chWidth) {
+				return;
+			}
+			if (mcs.isEmpty()) {
+				bitRate = rate.bitRate;
+				chWidth = rate.chWidth;
+			}
+			mcs.addAll(rate.mcs);
+		}
+	}
+
+	/**
+	 * Radio information with channel, channel_width and tx_power.
+	 */
+	public static class Radio {
+		public int channel;
+		public int channelWidth;
+		public int txPower;
+
+		private Radio() {}
+
+		public Radio(int channel, int channelWidth, int txPower) {
+			this.channel = channel;
+			this.channelWidth = channelWidth;
+			this.txPower = txPower;
+		}
+
+		private Radio(Map<String, Integer> radioInfo) {
+			channel = radioInfo.getOrDefault("channel", -1);
+			channelWidth = radioInfo.getOrDefault("channel_width", -1);
+			txPower = radioInfo.getOrDefault("tx_power", -1);
+		}
+
+		@Override
+		public int hashCode() {
+			return Objects.hash(channel, channelWidth, txPower);
+		}
+
+		@Override
+		public boolean equals(Object obj) {
+			if (this == obj) {
+				return true;
+			}
+			if (obj == null) {
+				return false;
+			}
+			if (getClass() != obj.getClass()) {
+				return false;
+			}
+
+			Radio other = (Radio) obj;
+			return channel == other.channel &&
+				channelWidth == other.channelWidth && txPower == other.txPower;
+		}
+	}
+
+	public String bssid;
+	public String station;
+	public long connected;
+	public long inactive;
+	public List<Integer> rssi;
+	public long rxBytes;
+	public long rxPackets;
+	public AggregatedRate rxRate;
+	public long txBytes;
+	public long txDuration;
+	public long txFailed;
+	public long txPackets;
+	public AggregatedRate txRate;
+	public long txRetries;
+	public int ackSignal;
+	public int ackSignalAvg;
+	public Radio radio;
+
+	/** Constructor with no args */
+	public AggregatedState() {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+		this.radio = new Radio();
+	}
+
+	/** Construct from Association and radio */
+	public AggregatedState(
+		Association association,
+		Map<String, Integer> radioInfo
+	) {
+		this.rxRate = new AggregatedRate();
+		this.txRate = new AggregatedRate();
+		this.rssi = new ArrayList<>();
+
+		this.bssid = association.bssid;
+		this.station = association.station;
+		this.connected = association.connected;
+		this.inactive = association.inactive;
+		this.rssi.add(association.rssi);
+		this.rxBytes = association.rx_bytes;
+		this.rxPackets = association.rx_packets;
+		this.rxRate.add(association.rx_rate);
+		this.txBytes = association.tx_bytes;
+		this.txDuration = association.tx_duration;
+		this.txFailed = association.tx_failed;
+		this.txPackets = association.tx_packets;
+		this.txRate.add(association.tx_rate);
+		this.txRetries = association.tx_retries;
+		this.ackSignal = association.ack_signal;
+		this.ackSignalAvg = association.ack_signal_avg;
+		this.radio = new Radio(radioInfo);
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(bssid, station, radio);
+	}
```

Comment:
You should NOT override `hashCode` differently from `equals`.... it will produce totally unexpected behavior unless you read this method (and if you did want to keep it, there should be a very large comment here explaining).